### ISSUE TITLE
"ignore: {}" interacts a bit strangely with custom hints

### ISSUE THREAD

**Issue by langston-barrett** [ndmitchell/hlint#747]
On this module:
```haskell
module Main where
main :: IO ()
main = undefined undefined
```
with this config:
```yaml
- ignore: {}
- error:
    name: "Only use one undefined"
    lhs: "undefined undefined"
    rhs: "undefined"
```
HLint 2.2.2 gives no hints. However, if I add this line:
```yaml
- error: {name: "Only use one undefined"}
```
then HLint raises the appropriate error. Without the `ignore: {}`, you don't need the second `error:` (and you don't have to name your hint).

This isn't a show-stopper, but is a bit unexpected.

**Comment by ndmitchell** [comment]
Agreed, this is unexpected. The error is defining a hint, but that is considered separate to the configuration. I'd accept a patch fixing this, but it's probably not trivial.

**Comment by neilmayhew** [comment]
This seems to have been fixed. With the original hints file and the test module, I get:

```
test-1.hs:3:8-26: Error: Only use one undefined
Found:
  undefined undefined
Perhaps:
  undefined

1 error
```

So I think this issue can be closed now.

**Comment by neilmayhew** [comment]
I got here from the [README](README.md#ignoring-hints) so that paragraph should probably be adjusted as well.

**Comment by langston-barrett** [comment]
Perhaps it would be good to add a regression test before closing this issue.
